src/app/blog/store/blog.effects.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from "@angular/core";
+import { Actions, createEffect, ofType  } from '@ngrx/effects';
+import * as BlogActions from "./blog.actions";
+import { catchError, map, of, switchMap } from "rxjs";
+import { PostsService } from "../services/posts.service";
+
+@Injectable()
+export class BlogEffects {
+
+    constructor(private actions$: Actions, private postsService: PostsService) {}      
+
+    fetchPosts$ = createEffect(() => 
```

Comment:
If you open comments and get back /posts request is sent again.

You may prevent it by doing something like:
```ts
  fetchPosts$ = createEffect(() =>
    this.actions$.pipe(
      ofType(BlogActions.fetchPosts),
      withLatestFrom(this.store.select(postsSelector)),
      switchMap(([action, storedPosts]) => {
        if (storedPosts.length > 0) return of(BlogActions.fetchPostsSuccess({posts: storedPosts}))
        return this.postsService.fetchPosts().pipe(
          map(posts => BlogActions.fetchPostsSuccess({posts})),
          catchError((error: Error) => of(BlogActions.fetchPostsFailure({errorMessage: error.message}))
          ));
      })));

Reply:
And in reducer:
```ts
    on(BlogActions.fetchPosts, (state) => {
        return { 
            ...state, 
            isPostsLoading: true,
        };
    }),